docs: clarify functionality of `$env/*` variables (#15078)
sveltejs/kit: packages/kit/src/types/synthetic/$env+dynamic+private.md

This module provides access to environment variables set _dynamically_ at runtime and that are limited to _private_ access.

|         | Runtime                                                                  | Buildtime                                                                |
| ------- | ------------------------------------------------------------------------ | ------------------------------------------------------------------------ |

> | Private | `$env/dynamic/private`                                                   | [`$env/static/private`](https://svelte.dev/docs/kit/$env-static-private) |

Reviewer: ```suggestion
| Private | [`$env/dynamic/private`](https://svelte.dev/docs/kit/$env-dynamic-private)                                                   | [`$env/static/private`](https://svelte.dev/docs/kit/$env-static-private) |
```

Reviewer: I know this is the current page but visually it's kind of confusing to have the others all be links and this one not

| Public  | [`$env/dynamic/public`](https://svelte.dev/docs/kit/$env-dynamic-public) | [`$env/static/public`](https://svelte.dev/docs/kit/$env-static-public)   |

> Runtime environment variables are defined by the platform you're running on. For example if you're using [`adapter-node`](https://github.com/sveltejs/kit/tree/main/packages/adapter-node) (or running [`vite preview`](https://svelte.dev/docs/kit/cli)), this is equivalent to `process.env`.

Reviewer: ```suggestion
Dynamic environment variables are defined by the platform you're running on. For example if you're using [`adapter-node`](https://github.com/sveltejs/kit/tree/main/packages/adapter-node) (or running [`vite preview`](https://svelte.dev/docs/kit/cli)), this is equivalent to `process.env`.
```

**_Private_ access:**

- This module (and [`$env/static/private`](https://svelte.dev/docs/kit/$env-static-private)) cannot be imported into client-side code.
- Variables that begin with [`config.kit.env.publicPrefix`](https://svelte.dev/docs/kit/configuration#env) (which defaults to `PUBLIC_`) are **excluded**.

> - Variables that begin with [`config.kit.env.privatePrefix`](https://svelte.dev/docs/kit/configuration#env) (which defaults to allow-all) are **included**.

Reviewer: ```suggestion
- Variables that begin with [`config.kit.env.privatePrefix`](https://svelte.dev/docs/kit/configuration#env) (which defaults to `''`, allowing everything) are **included**.
```

> - Variables that begin with [`config.kit.env.privatePrefix`](https://svelte.dev/docs/kit/configuration#env) (which defaults to allow-all) are **included**.

Reviewer: The original verbage was more correct:

```suggestion
- This module includes variables that _do not_ begin with [`config.kit.env.publicPrefix`](https://svelte.dev/docs/kit/configuration#env) _and do_ start with [`config.kit.env.privatePrefix`](https://svelte.dev/docs/kit/configuration#env) (if configured)
```

> [!NOTE] In `dev`, `$env/dynamic` includes environment variables from `.env`. In `prod`, this behavior will depend on your adapter.

> [!NOTE] To get correct types, environment variables referenced in your code should be declared (for example in an `.env` file), even if they don't have a value until the app is deployed:
>
> ```env
> MY_FEATURE_FLAG=
> ```
>
> You can override `.env` values from the command line like so:
>
> ```sh
> MY_FEATURE_FLAG="enabled" npm run dev
> ```

> For example, suppose the runtime environment variables were set like this:

Reviewer: ```suggestion
For example, given the following `.env`:
```
